### DISCUSSION TITLE
Workflow not showing and not triggered by release

### DISCUSSION THREAD

**Discussion by tcurdt** [community/community#26875] [Actions]
<p>I have a workflow creating a release for me</p>
<pre><code class="lang-auto">
  release:
    if: github.event_name == 'push' &amp;&amp; startsWith(github.ref, 'refs/tags/v')
    needs: [test]
    runs-on: ubuntu-latest
    steps:

      ...

      - name: Release
        uses: softprops/action-gh-release@v1
        with:
          files: |
            artifacts/*.tgz
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>Now I have another workflow that should publish that release somewhere else.</p>
<pre><code class="lang-auto">on:
  release:
    types: [published]

jobs:
...
</code></pre>
<p>But this workflow does not get triggered.<br>
It does not even show up under “actions &gt; All Workflows”</p>
<p>What am I missing here?</p>

**Comment by tcurdt** [comment]
<p>Creating the release from the UI does trigger the workflow.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/tcurdt">@tcurdt</a> ,</p>
<p>According to the introduction from the docs about the <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow" rel="noopener nofollow ugc"><strong>GITHUB_TOKEN</strong></a>:</p>
<blockquote>
<p>When you use the repository’s <strong>GITHUB_TOKEN</strong> to perform tasks on behalf of the GitHub Actions app, events triggered by the <strong>GITHUB_TOKEN</strong> will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.</p>
</blockquote>
<p>Due to you are using the <strong>GITHUB_TOKEN</strong> to authenticate on the action to create the new release, the workflow can’t be triggered on the <strong>release</strong> event.</p>
<p>If you want the workflow can be triggered when creating a new release via the action, you need to create a personal access token (PAT) and set it as a secret in your repository. Then use this PAT to authenticate on the action. More details, see “<a href="https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token" rel="noopener nofollow ugc"><strong>Creating a personal access token</strong></a>”.</p>
<br>
<blockquote>
<p>It does not even show up under “actions &gt; All Workflows”</p>
</blockquote>
<p>Generally, if the workflow file does not exist on the default or master branch, this workflow will not be automatically listed under “<strong>Actions</strong> &gt; <strong>All workflows</strong>” unless this workflow has been triggered at least once.</p>

**Reply by DiFuks** [reply]
Everything works with the PAT, but another problem arises. I create a new Github Release in my workflow. But in this case, I always end up being the author of the release, since the PAT belongs to me.

**Comment by tcurdt** [comment]
<blockquote>
<p>When you use the repository’s <strong>GITHUB_TOKEN</strong> to perform tasks on behalf of the GitHub Actions app, events triggered by the <strong>GITHUB_TOKEN</strong> will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.</p>
</blockquote>
<p>I wish this was linked from the docs about the triggers. I didn’t see it.<br>
I wish the github team responsible would have used a different approach to prevent recursive workflows (if really deemed unacceptable).<br>
I also wish the github team responsible would have though that this might cause problems and at least give an error message why the workflow wasn’t triggered.</p>
<p>Thank you so much for the pointer. I will give that a try today.</p>
<blockquote>
<p>Generally, if the workflow file does not exist on the default or master branch, this workflow will not be automatically listed under “ <strong>Actions</strong> &gt; <strong>All workflows</strong> ” unless this workflow has been triggered at least once.</p>
</blockquote>
<p>It is on the master branch. So why doesn’t it show?</p>
<p>I have also added another workflow - that uses the same trigger. That one does show up.</p>

**Comment by tcurdt** [comment]
<p>Using the PAT all seems to be working. Thanks for the help.</p>
